docs/reference/ingest/processors/json.asciidoc

```diff
@@ -10,10 +10,11 @@ Converts a JSON string into a structured JSON object.
 .Json Options
 [options="header"]
 |======
-| Name           | Required  | Default  | Description
-| `field`        | yes       | -        | The field to be parsed.
-| `target_field` | no        | `field`  | The field that the converted structured object will be written into. Any existing content in this field will be overwritten.
-| `add_to_root`  | no        | false    | Flag that forces the serialized json to be injected into the top level of the document. `target_field` must not be set when this option is chosen.
+| Name                         | Required  | Default  | Description
+| `field`                      | yes       | -        | The field to be parsed.
+| `target_field`               | no        | `field`  | The field that the converted structured object will be written into. Any existing content in this field will be overwritten.
+| `add_to_root`                | no        | false    | Flag that forces the serialized json to be merged into the top level of the document. `target_field` must not be set when this option is chosen.
+| `add_to_root_recursive_merge`| no        | false    | Flag that controls the merging strategy for `add_to_root`. When set to `false`, only the first level will be merged. When set to `true`, even nested objects will be merged. Only applicable if `add_to_root` is set to `true`.
```

Comment:
I'm not so sure about the name. Not sure what a better name is. Maybe `add_to_root_merge_strategy` with values `recursive` and `replace`? I curious to see what others think.

Reply:
+1 I like the descriptiveness of the values.
I've created another PR that implements that: https://github.com/elastic/elasticsearch/pull/74967. Feel free to close this PR in favor of the `add_to_root_merge_strategy` PR.